libp2p/protocols/pubsub/floodsub.nim

```diff
@@ -65,9 +65,13 @@ method rpcHandler*(f: FloodSub,
         if msg.msgId notin f.seen:
           f.seen.put(msg.msgId)                      # add the message to the seen cache
 
-          if f.verifySignature and not msg.verify(peer.peerInfo):
-            trace "dropping message due to failed signature verification"
-            continue
+          if f.verifySignature:
+            if not msg.verify(peer.peerInfo):
+              libp2p_pubsub_sig_verify_failure.inc()
```

Comment:
probably better to move this inside `msg.verify`, that way we don't have to duplicate if we ever use validation on a protocol that doesn't derive from floodsub.

Reply:
Ah that's a very good point. Thought it was a crypto proc, but either way better generalize I guess.